### PR TITLE
Improve schedule, and show schedule in trip show

### PR DESCRIPTION
**Pull request by Producermark**
Example:

`{
    "schedule": [
        {
            "day": "unsorted",
            "items": []
        },
        {
            "day": 1,
            "items": [
                {
                    "id": 1,
                    "venue": {
                        "id": "43222200f964a5209c271fe3",
                        "name": "Gantry Plaza State Park",
                        "description": "",
                        "address": [
                            "474 48th Ave (at Center Blvd)",
                            "Long Island City, NY 11109",
                            "United States"
                        ],
                        "latitude": 40.74655794031174,
                        "longitude": -73.9580512046814,
                        "categories": [
                            {
                                "name": "Park",
                                "pluralName": "Parks",
                                "icon": "https://ss3.4sqi.net/img/categories_v2/parks_outdoors/park_64.png"
                            },
                            {
                                "name": "Scenic Lookout",
                                "pluralName": "Scenic Lookouts",
                                "icon": "https://ss3.4sqi.net/img/categories_v2/parks_outdoors/sceniclookout_64.png"
                            }
                        ],
                        "images": [
                            {
                                "shotAt": "Saturday, 22-Sep-2012 21:27:37 UTC",
                                "photographer": "Andrew Deegan",
                                "photographerImage": "https://igx.4sqi.net/img/user/250x250/XNJROOLAY4HPCAYL.jpg",
                                "horizontalRectangleURL": "https://igx.4sqi.net/img/general/800x344/3868739_NSiIiahz4k3T84CMeH7olM18Iyl4RvmjjKj0O29-8VU.jpg",
                                "squareURL": "https://igx.4sqi.net/img/general/612x612/3868739_NSiIiahz4k3T84CMeH7olM18Iyl4RvmjjKj0O29-8VU.jpg"
                            }
                        ],
                        "url": "http://nysparks.com/parks/149/details.aspx",
                        "openingHours": null,
                        "popularHours": null,
                        "price": null,
                        "rating": "9.6",
                        "ratingColor": "00B551",
                        "peopleNow": "0",
                        "likes": 688
                    }
                },
                {
                    "id": 3,
                    "venue": {
                        "id": "430d0a00f964a5203e271fe3",
                        "name": "Brooklyn Bridge Park",
                        "description": "",
                        "address": [
                            "Main St (Plymouth St)",
                            "Brooklyn, NY 11201",
                            "United States"
                        ],
                        "latitude": 40.702282,
                        "longitude": -73.996456,
                        "categories": [
                            {
                                "name": "Park",
                                "pluralName": "Parks",
                                "icon": "https://ss3.4sqi.net/img/categories_v2/parks_outdoors/park_64.png"
                            }
                        ],
                        "images": [
                            {
                                "shotAt": "Saturday, 04-May-2013 12:17:33 UTC",
                                "photographer": "🐾Daria Spiridonova",
                                "photographerImage": "https://igx.4sqi.net/img/user/250x250/46606820-IVF3F4KAUDPIJBA4.jpg",
                                "horizontalRectangleURL": "https://igx.4sqi.net/img/general/800x344/46606820_YgjTokGqRcelaf2Pc86R8yID8ufSZC4h-LxFq0LwoXg.jpg",
                                "squareURL": "https://igx.4sqi.net/img/general/1440x1440/46606820_YgjTokGqRcelaf2Pc86R8yID8ufSZC4h-LxFq0LwoXg.jpg"
                            }
                        ],
                        "url": "http://nyc.gov/parks",
                        "openingHours": null,
                        "popularHours": null,
                        "price": null,
                        "rating": "9.7",
                        "ratingColor": "00B551",
                        "peopleNow": "0",
                        "likes": 3641
                    }
                }
            ]
        },
        {
            "day": 2,
            "items": [
                {
                    "id": 2,
                    "venue": {
                        "id": "412d2800f964a520df0c1fe3",
                        "name": "Central Park",
                        "description": "Central Park is the 843-acre green heart of Manhattan and is maintained by the Central Park Conservancy. It was designed in the 19th century by Frederick Law Olmsted and Calvert Vaux as an urban escape for New Yorkers, and now receives over 40 million visits per year.",
                        "address": [
                            "59th St to 110th St (5th Ave to Central Park West)",
                            "New York, NY 10028",
                            "United States"
                        ],
                        "latitude": 40.78408342593807,
                        "longitude": -73.96485328674316,
                        "categories": [
                            {
                                "name": "Park",
                                "pluralName": "Parks",
                                "icon": "https://ss3.4sqi.net/img/categories_v2/parks_outdoors/park_64.png"
                            }
                        ],
                        "images": [
                            {
                                "shotAt": "Sunday, 10-Mar-2013 00:21:55 UTC",
                                "photographer": "Benjamin Bannister",
                                "photographerImage": "https://igx.4sqi.net/img/user/250x250/655018-SOLGBBXDKHVPRIPM.jpg",
                                "horizontalRectangleURL": "https://igx.4sqi.net/img/general/800x344/655018_Zp3vA90Sy4IIDApvfAo5KnDItoV0uEDZeST7bWT-qzk.jpg",
                                "squareURL": "https://igx.4sqi.net/img/general/612x612/655018_Zp3vA90Sy4IIDApvfAo5KnDItoV0uEDZeST7bWT-qzk.jpg"
                            }
                        ],
                        "url": "http://www.centralparknyc.org",
                        "openingHours": "Closed until 6:00 AM",
                        "popularHours": null,
                        "price": null,
                        "rating": "9.8",
                        "ratingColor": "00B551",
                        "peopleNow": "0",
                        "likes": 20260
                    }
                },
                {
                    "id": 5,
                    "venue": {
                        "id": "427c0500f964a52097211fe3",
                        "name": "The Metropolitan Museum of Art (Metropolitan Museum of Art)",
                        "description": "The Metropolitan Museum of Art is the largest art museum in the United States with among the most significant art collections. Its permanent collection contains more than two million works, divided among nineteen curatorial departments.",
                        "address": [
                            "1000 5th Ave (btwn E 80th & E 84th St)",
                            "New York, NY 10028",
                            "United States"
                        ],
                        "latitude": 40.77972902126812,
                        "longitude": -73.96341562271118,
                        "categories": [
                            {
                                "name": "Art Museum",
                                "pluralName": "Art Museums",
                                "icon": "https://ss3.4sqi.net/img/categories_v2/arts_entertainment/museum_art_64.png"
                            }
                        ],
                        "images": [
                            {
                                "shotAt": "Monday, 15-Jul-2013 15:28:28 UTC",
                                "photographer": "Chayapol Kongton",
                                "photographerImage": "https://igx.4sqi.net/img/user/250x250/YXNQ5O4TLYSNHMKY.jpg",
                                "horizontalRectangleURL": "https://igx.4sqi.net/img/general/800x344/1022386_wEd7VXnWzp5lajvRLtAPunoiufDSIq8PMEtEuIH2Tzg.jpg",
                                "squareURL": "https://igx.4sqi.net/img/general/612x612/1022386_wEd7VXnWzp5lajvRLtAPunoiufDSIq8PMEtEuIH2Tzg.jpg"
                            }
                        ],
                        "url": "http://www.metmuseum.org",
                        "openingHours": "Closed until 10:00 AM",
                        "popularHours": null,
                        "price": null,
                        "rating": "9.7",
                        "ratingColor": "00B551",
                        "peopleNow": "0",
                        "likes": 13707
                    }
                }
            ]
        },
        {
            "day": 3,
            "items": [
                {
                    "id": 4,
                    "venue": {
                        "id": "4ec0a81fb6341cd41f1ef402",
                        "name": "Minskoff Theatre",
                        "description": "",
                        "address": [
                            "1515 Broadway (at W 45th St)",
                            "New York, NY 10036",
                            "United States"
                        ],
                        "latitude": 40.75749442223194,
                        "longitude": -73.98568022741497,
                        "categories": [
                            {
                                "name": "Theater",
                                "pluralName": "Theaters",
                                "icon": "https://ss3.4sqi.net/img/categories_v2/arts_entertainment/performingarts_theater_64.png"
                            }
                        ],
                        "images": [
                            {
                                "shotAt": "Thursday, 27-Jun-2013 23:49:31 UTC",
                                "photographer": "Korinne Caceres",
                                "photographerImage": "https://igx.4sqi.net/img/user/250x250/Z4RUMVL11TYM3N2M.jpg",
                                "horizontalRectangleURL": "https://igx.4sqi.net/img/general/800x344/1577063_cm4AqdTUAE9GfkqAyivqyfX6QWVE69mY6jwod8-uMBc.jpg",
                                "squareURL": "https://igx.4sqi.net/img/general/720x720/1577063_cm4AqdTUAE9GfkqAyivqyfX6QWVE69mY6jwod8-uMBc.jpg"
                            }
                        ],
                        "url": "http://www.minskofftheatre.com",
                        "openingHours": null,
                        "popularHours": null,
                        "price": null,
                        "rating": "9.7",
                        "ratingColor": "00B551",
                        "peopleNow": "0",
                        "likes": 1273
                    }
                },
                {
                    "id": 6,
                    "venue": {
                        "id": "4a66f31af964a520dcc81fe3",
                        "name": "Equinox West 50th Street",
                        "description": "Equinox isn't just a fitness club, it's a temple of well-being.  Discover an unparalleled member experience where innovative programming,  rejuvenating amenities and renowned instruction come together to create extraordinary results.",
                        "address": [
                            "1633 Broadway (at Paramount Plz)",
                            "New York, NY 10019",
                            "United States"
                        ],
                        "latitude": 40.761912,
                        "longitude": -73.984268,
                        "categories": [
                            {
                                "name": "Gym",
                                "pluralName": "Gyms",
                                "icon": "https://ss3.4sqi.net/img/categories_v2/building/gym_64.png"
                            },
                            {
                                "name": "Gym / Fitness Center",
                                "pluralName": "Gyms or Fitness Centers",
                                "icon": "https://ss3.4sqi.net/img/categories_v2/building/gym_64.png"
                            }
                        ],
                        "images": [
                            {
                                "shotAt": "Friday, 18-Aug-2017 20:47:46 UTC",
                                "photographer": "Pivot_photos",
                                "photographerImage": "https://igx.4sqi.net/img/user/250x250/blank_boy.png",
                                "horizontalRectangleURL": "https://igx.4sqi.net/img/general/800x344/212169387_C2-Wh9iB-a9yFzkkMFUuT3RsxV7Rna5lQWhqCNs0Bds.jpg",
                                "squareURL": "https://igx.4sqi.net/img/general/2592x2592/212169387_C2-Wh9iB-a9yFzkkMFUuT3RsxV7Rna5lQWhqCNs0Bds.jpg"
                            }
                        ],
                        "url": "http://www.facebook.com/equinox",
                        "openingHours": "Closed until 5:30 AM",
                        "popularHours": null,
                        "price": null,
                        "rating": "9.7",
                        "ratingColor": "00B551",
                        "peopleNow": "0",
                        "likes": 212
                    }
                },
                {
                    "id": 7,
                    "venue": {
                        "id": "3fd66200f964a520d7f11ee3",
                        "name": "Bryant Park",
                        "description": "Located in Midtown Manhattan, Bryant Park is visited by over 6 million people each year and is one of the busiest public spaces in the world. Check out everything you can do in the park year round at bryantpark.org!",
                        "address": [
                            "E 42nd St (btwn 5th & 6th Ave)",
                            "New York, NY 10018",
                            "United States"
                        ],
                        "latitude": 40.75395491241381,
                        "longitude": -73.98398145637879,
                        "categories": [
                            {
                                "name": "Park",
                                "pluralName": "Parks",
                                "icon": "https://ss3.4sqi.net/img/categories_v2/parks_outdoors/park_64.png"
                            }
                        ],
                        "images": [
                            {
                                "shotAt": "Wednesday, 12-Jun-2013 19:38:44 UTC",
                                "photographer": "Alexandra Tims",
                                "photographerImage": "https://igx.4sqi.net/img/user/250x250/ZUU1XZG4ND2JFQEI.png",
                                "horizontalRectangleURL": "https://igx.4sqi.net/img/general/800x344/1112348_XN9BG-MFXC20fhjZAXNtGeEd8kzgFOWQ7n8Q7LsKEvE.jpg",
                                "squareURL": "https://igx.4sqi.net/img/general/612x612/1112348_XN9BG-MFXC20fhjZAXNtGeEd8kzgFOWQ7n8Q7LsKEvE.jpg"
                            }
                        ],
                        "url": "http://www.facebook.com/bryantparknyc",
                        "openingHours": null,
                        "popularHours": null,
                        "price": null,
                        "rating": "9.7",
                        "ratingColor": "00B551",
                        "peopleNow": "1",
                        "likes": 9873
                    }
                }
            ]
        }
    ]
}`

### CODE REVIEW

File: backend/app/ContentProviders/FourSquare/VenueFactory.php [hunk 1]
```diff
@@ -160,7 +160,10 @@ private function setVenueImages($images, Venue $venue)
         foreach($images as $image) {
             $newImage = new VenueImage();
             $newImage->setShotAt(\Carbon\Carbon::createFromTimestamp($image->createdAt)->toCookieString());
-            $newImage->setPhotographer(ucfirst($image->user->firstName) . " " . ucfirst($image->user->lastName));
+            if(isset($image->user->lastName))
+                $newImage->setPhotographer(ucfirst($image->user->firstName) . " " . ucfirst($image->user->lastName));
+            else
```
Comment (killje) [inline backend/app/ContentProviders/FourSquare/VenueFactory.php]: This should be in pr #69

File: backend/app/Http/Controllers/API/TripController.php [hunk 2]
```diff
@@ -208,7 +202,7 @@ public function show(Request $request, ContentProvider $contentProvider)
                 'created_by' => $trip->created_by,
                 'created_at' => $trip->created_at->toCookieString(),
                 'updated_at' => $trip->updated_at->toCookieString(),
-                'venues' => $venuesJSON
+                'schedule' => $trip->generateSchedule($contentProvider, false, true),
```
Comment (killje) [inline backend/app/Http/Controllers/API/TripController.php]: Does this not generate a new schedule instead of using the existing scedule?

Comment (Producermark) [reply backend/app/Http/Controllers/API/TripController.php]: No, that's why it sends false as an argument

File: backend/app/ContentProviders/FourSquare/VenueFactory.php (hunk 1 above)
Comment (Producermark) [reply backend/app/ContentProviders/FourSquare/VenueFactory.php]: It will be, if this gets merged (that's why it's a pr into #69)

**Pull request merged**